GitHubContents.ps1

```diff
@@ -0,0 +1,122 @@
+        returned as a string instead of bytes. If the MediaType is Object, then an additional property on the object is returned 'contentAsString'
+        which will be the decoded base64 result as a string.
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Get-GitHubContent -OwnerName microsoft -RepositoryName PowerShellForGitHub -Path README.md -MediaType Html
+
+        Get the Html output for the README.md file
+
+    .EXAMPLE
+        Get-GitHubContent -OwnerName microsoft -RepositoryName PowerShellForGitHub -Path LICENSE
+
+        Get the Binary file output for the LICENSE file
+
+    .EXAMPLE
+        Get-GitHubContent -OwnerName microsoft -RepositoryName PowerShellForGitHub -Path Tests
+
+        List the files within the "Tests" path of the repository
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParametersetName = 'Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification = "Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(Mandatory, ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(Mandatory, ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [string] $Path,
+
+        [ValidateSet('Raw', 'Html', 'Object')]
+        [string] $MediaType = 'Object',
+
+        [switch] $ResultAsString,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements -DisableValidation
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName'      = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $uriFragment = [String]::Empty
+    $description = [String]::Empty
+
+    $uriFragment = "/repos/$OwnerName/$RepositoryName/contents"
+
+    if ($PSBoundParameters.ContainsKey('Path')) {
+        $Path = $Path.TrimStart("\", "/")
+        $uriFragment += "/$Path"
+        $description = "Getting content for $Path in $RepositoryName"
+    }
+    else {
+        $description = "Getting all content for in $RepositoryName"
+    }
+
+    $params = @{
+        'UriFragment'         = $uriFragment
```

Comment:
formatting: Here and above (`$telemetryProperties`) - This project doesn't attempt to line up values this way.